types/node/test/stream.ts

```diff
@@ -4,12 +4,17 @@ import { createReadStream, createWriteStream } from 'node:fs';
 import { createGzip, constants } from 'node:zlib';
 import assert = require('node:assert');
 import { Http2ServerResponse } from 'node:http2';
-import { text, json, buffer } from 'node:stream/consumers';
+import { text, json, buffer, arrayBuffer, blob } from 'node:stream/consumers';
 import { pipeline as pipelinePromise } from 'node:stream/promises';
 import { stdout } from 'node:process';
 import { ReadableStream, WritableStream, TransformStream } from 'node:stream/web';
 import { setInterval as every } from 'node:timers/promises';
 import { MessageChannel } from 'node:worker_threads';
+import { performance } from 'node:perf_hooks';
+
+// Ensure there is no global Blob type
+// $ExpectError
+type ShouldFail = Blob;
```

Comment:
Node.js 18.0.0 has a global Blob for me 🤔 

<img width="232" alt="Screenshot 2022-07-07 at 18 09 16" src="https://user-images.githubusercontent.com/189580/177820577-ef95ddab-ca76-4480-b71b-1fa0a8b5abed.png">

Reply:
I can't find any docs for this.  Do you know when it was introduced?  I'm still on an older LTS (14.7 I think? Don't update that often.) which definitely does not have a global Blob.

Can you please run `Blob === buffer.Blob` for me and see what it says?  May need to `import buffer from 'buffer'` first.

Reply:
So, it looks like as of Node v18, Blob is moved/aliased into the global namespace intentionally, which negates a big part of the motivation for this PR. The problem is, Node's Blob must merge correctly with the one in lib-dom, but can't reference it directly unless/until we get something like [this feature](https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1207), which hasn't seen any movement in a while.

I'm not sure how to proceed.  The motivating case is still an issue -- projects that include both `@types/node` and `lib: ['dom']` can still see buggy behavior.  Maybe some of the core `@types/node` authors could weigh in here?

Reply:
I should add that I think that this PR is a really good idea (with my limited understanding) since most dom apis are _not_ available in Node.js, and having it include `dom` isn't super...

Reply:
<img width="337" alt="Screenshot 2022-07-08 at 14 58 37" src="https://user-images.githubusercontent.com/189580/177996629-f1ab46ee-5c52-4995-b85f-454e221bae4e.png">

Reply:
Per the discussion below I'm suggesting globally exposing the Blob constructor, which we can do conditionally, but not the Blob type, which we can't.  Resolving this, since it's part of the larger discussion below.